jaq-native/src/math.rs

```diff
@@ -0,0 +1,108 @@
+use alloc::string::String;
+use jaq_core::{Error, Val};
+
+/// Use a value as a float to be given as an argument to a libm
+/// function.
+pub fn as_float(v: Val, fun: &str) -> Result<f64, Error> {
+    match v {
+        Val::Int(n) => Ok(n as f64),
+        Val::Float(n) => Ok(n),
+        Val::Num(ref n) => n.parse().or(Err(Error::MathFn(String::from(fun), v))),
+        _ => Err(Error::MathFn(String::from(fun), v)),
+    }
+}
+
+/// Use a value as an integer to be given as an argument to a libm
+/// function.
+pub fn as_int(v: Val, fun: &str) -> Result<i32, Error> {
+    match v {
+        Val::Int(n) => Ok(n as i32),

```

Comment:
This looks dangerous -- it looks like it could lose information if `isize` is 64 bits.
Better do a `n.try_into()` and fail with error if it fails.